pytorch_lightning/utilities/data.py

```diff
@@ -197,38 +196,71 @@ def _update_dataloader(
             f" can be passed to its parent's `{argument}=...` `__init__` argument. This is likely caused by allowing"
             f" passing both a custom argument that will map to the `{argument}` argument as well as `**kwargs`."
             f" `kwargs` should be filtered to make sure they don't contain the `{argument}` key."
+            " This argument was automatically passed to your DataLoader by PyTorch Lightning."
         )
         raise MisconfigurationException(message) from e
     return dataloader
 
 
-def _get_dataloader_init_kwargs(
+def _get_dataloader_init_args_and_kwargs(
     dataloader: DataLoader, sampler: Optional[Sampler], mode: Optional[RunningStage] = None
-) -> Dict[str, Any]:
+) -> Tuple[Tuple[Any], Dict[str, Any]]:
     if not isinstance(dataloader, DataLoader):
         raise ValueError(f"The dataloader {dataloader} needs to subclass `torch.utils.data.DataLoader`")
 
-    # get the dataloader instance attributes
-    attrs = {k: v for k, v in vars(dataloader).items() if not k.startswith("_")}
-    # not part of `vars`
-    attrs["multiprocessing_context"] = dataloader.multiprocessing_context
+    was_wrapped = hasattr(dataloader, "__pl_dl_args")
+    if was_wrapped:
+        args = dataloader.__pl_dl_args
+        attrs = dataloader.__pl_dl_kwargs
+        arg_names = dataloader.__pl_dl_arg_names
+        used_args = min(len(args), len(arg_names))
+        for arg, arg_name in zip(args, arg_names):
```

Comment:
super nit: keys are usually in the left

```suggestion
        for arg_name, arg in zip(arg_names, args):
```

Reply:
Not there anymore